File: app/javascript/channels/user_channel.js
import consumer from "./consumer"

const subscribeChannel = (userId) => {
  consumer.subscriptions.create({
    channel: 'UserChannel',
    user_id: userId
  }, {
    received(data) {
      console.log(data);
    }
  });
}

// const subscribeChannel = (conversationId) => {
//   consumer.subscriptions.create({
//     channel: 'ConversationChannel',
//     conversation_id: conversationId
//   }, {
//     received(data) {
//       console.log(data);
//     }
//   });
// }

export default subscribeChannel;

File: app/channels/user_channel.rb
class UserChannel < ApplicationCable::Channel
  def subscribed
    stream_from "user_#{params[:user_id]}"
  end
end


# class ConversationChannel < ApplicationCable::Channel
#   def subscribed
#     stream_from "conversation_#{params[:conversation_id]}"
#   end
# end
